src/Integration/Flysystem/S3/tests/Unit/S3FilesystemV1Test.php

```diff
@@ -39,7 +40,7 @@ public function testWrite()
 
         $result = $this->getMockBuilder(PutObjectOutput::class)
```

Comment:
What's about using the `ResultMockFactory`?

Reply:
I forgot about ResultMockFactory =)

But I still have issues with pagination.. I'll see if I can solve it.

Reply:
If you can't, Im fine with move the final from method to class